src/main/scala/org/ergoplatform/nodeView/wallet/Registry.scala

```diff
@@ -0,0 +1,138 @@
+package org.ergoplatform.nodeView.wallet
+
+import io.iohk.iodb.ByteArrayWrapper
+import org.ergoplatform.nodeView.history.ErgoHistory.Height
+
+import scala.collection.mutable
+
+object Registry {
+
+  private val registry = mutable.Map[ByteArrayWrapper, TrackedBox]()
+
+  private val confirmedIndex = mutable.TreeMap[Height, Seq[ByteArrayWrapper]]()
```

Comment:
here as well: we need a concurrent data structure, or all reads and writes should be synchronized